src/bbs_verify.sol

```diff
@@ -0,0 +1,705 @@
+
+        // Step 2: Add uncompressed G1 point q1
+        domOcts = abi.encodePacked(domOcts, g1ToBytes(BBS.generators()[0]));
+
+        // Step 3: Add each hPoint uncompressed
+        for (uint256 i = 1; i < h_points_len + 1; i++) {
+            domOcts = abi.encodePacked(domOcts, g1ToBytes(BBS.generators()[i]));
+        }
+
+        // Step 4: Add the API ID
+        domOcts = abi.encodePacked(domOcts, api_id);
+
+        // Step 5: Add compressed G2 public key
+        bytes memory x1Bytes = reverseBytes(uintToBytes(pk.PK.X[1]));
+        bytes memory x0Bytes = reverseBytes(uintToBytes(pk.PK.X[0]));
+        bytes memory y1Bytes = reverseBytes(uintToBytes(pk.PK.Y[1]));
+        bytes memory y0Bytes = reverseBytes(uintToBytes(pk.PK.Y[0]));
+        bytes memory compressedPk = abi.encodePacked(x1Bytes, x0Bytes, y1Bytes, y0Bytes);
+
+        // Step 6: Create final domain input
+        bytes memory domInput = abi.encodePacked(compressedPk, domOcts);
+
+        // Step 7: Add header length (big-endian 8 bytes) and header
+        bytes1 zeroByte = 0x00;
+        domInput =
+            abi.encodePacked(domInput, zeroByte, zeroByte, zeroByte, zeroByte, zeroByte, zeroByte, zeroByte, zeroByte);
+
+        // Step 8: Perform hash-to-scalar
+        return BBS.hashToScalar(domInput, dst);
+    }
+
+    // Helper function to convert uint64 to bytes (big-endian)
+    function uint64ToBytes(uint256 x) internal pure returns (bytes memory) {
+        bytes memory result = new bytes(8); // uint64 takes 8 bytes
+        assembly {
+            let resultPtr := add(result, 32)
+            mstore(resultPtr, shl(192, x))
+        }
+        return result;
+    }
+
+    function flag(uint256 y) internal pure returns (bool) {
+        if (y <= Pairing.PRIME_Q - y) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // Helper function to convert G1 point to bytes
+    function g1ToBytes(Pairing.G1Point memory point) public pure returns (bytes memory) {
+        bytes memory xBytes = reverseBytes(uintToBytes(point.X));
+        bytes memory yBytes = reverseBytes(uintToBytes(point.Y));
+        if (!flag(point.Y)) {
+            yBytes[31] = bytes1(uint8(yBytes[31]) | uint8(1 << 7));
+        }
+        return abi.encodePacked(xBytes, yBytes);
+    }
+
+    function uintToBytes(uint256 x) public pure returns (bytes memory) {
+        bytes memory b = new bytes(32); // A uint256 is always 32 bytes
+        assembly {
+            mstore(add(b, 32), x) // Store x into the bytes array
+        }
+        return b;
+    }
+
+    function reverseBytes(bytes memory input) internal pure returns (bytes memory) {
+        bytes memory output = new bytes(32);
+
+        for (uint256 i = 0; i < (input.length + 1) / 2; i++) {
+            output[i] = input[input.length - 1 - i];
+            output[input.length - 1 - i] = input[i];
+        }
+
+        return output;
+    }
+
+    function proofVerify(
```

Comment:
Keep the naming consistent. So, this should be `verifyProof`

Reply:
fixed